### PR TITLE
feat: #24670 allow user customize table font size with less variable

### LINKED ISSUE #24670: Add a less default variables that allows to control the table-font-size
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
The variables allow changing table font size depending on ConfigProvider's componentSize.

### What does the proposed API look like?
```
@table-font-size-base: 14px;
@table-font-size-lg: @table-font-size-base + 2px;
@table-font-size-sm: 12px;
```

### PR DESCRIPTION
**Pull request by morenyang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

close #24670 

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/24670

### 💡 Background and solution

- add variable `@table-font-size`, `@table-font-size-md`, `@table-font-size-sm`
- to avoid impact user's style, these three variables are set to `@font-size-base` by default

Demo on local (set `font-size` to 16px, 14px, 12px temporary)
ezgif-2-595877b32398 (1)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add less variable `@table-font-size`, `@table-font-size-md` and `@table-font-size-sm` to customize the size of Table text. |
| 🇨🇳 Chinese | 新增 less 变量 `@table-font-size`，`@table-font-size-md` 和 `@table-font-size-sm`，用于自定义 Table 文字大小。 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/style/themes/default.less [hunk 1]
```diff
@@ -574,6 +574,9 @@
 @table-footer-bg: @background-color-light;
 @table-footer-color: @heading-color;
 @table-header-bg-sm: @table-header-bg;
+@table-font-size: @font-size-base;
```
Comment (xrkffgg) [inline components/style/themes/default.less:577]: 关注下 紧凑模式  themes/compact.less 是不是需要调整。

Comment (xrkffgg) [inline components/style/themes/default.less:577]: 关注下 紧凑模式  themes/compact.less 是不是需要调整。

@table-font-size-md  @table-font-size-sm  都用 base 吗？

Comment (xrkffgg) [reply components/style/themes/default.less:577]: 我看你的 background 写的是3个尺寸呀

Comment (morenyang) [reply components/style/themes/default.less:577]: 我觉得我们只是提供一个让用户能自定义的方式，如果直接改的话会影响已有的样式

**Pull request merged**